### PR TITLE
Add asahi-btsync

### PR DESCRIPTION
**Pull request by jannau**
Parses "BluetoothUHEDevices" and creates stub info files with a pairing keys.

Signed-off-by: Janne Grunau <j@jannau.net>

### CODE REVIEW

File: asahi-btsync/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+    let bt_var = "BluetoothUHEDevices";
+
+    let mut file = OpenOptions::new()
+        .read(true)
+        .open(matches.get_one::<String>("device").unwrap_or(&default_name))
+        .unwrap();
+    let mut data = Vec::new();
+    file.read_to_end(&mut data).unwrap();
+    let mut nv = Nvram::parse(&data)?;
+    let bt_devs = part_by_name("system", &mut nv)?
```
Comment (WhatAmISupposedToPutHere) [inline asahi-btsync/src/main.rs:78]: I do not think you need `part_by_name` here, just access the system partition directly

File: asahi-btsync/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+    let matches = clap::command!()
+        .arg(clap::arg!(-d --device [DEVICE] "Path to the nvram device."))
+        .subcommand(clap::Command::new("list").about("Parse shared Bluetooth keys from nvram"))
+        .subcommand(
+            clap::Command::new("sync")
+                .about("Sync Bluetooth device information from nvram")
+                .arg(clap::arg!(-c --config [CONFIG] "Bluez config path."))
+                .arg(clap::Arg::new("variable").multiple_values(true)),
+        )
+        .subcommand(clap::Command::new("dump").about("Dump binary Bluetooth device info frm nvram"))
```
Comment (WhatAmISupposedToPutHere) [inline asahi-btsync/src/main.rs:64]: from nvram

**Review by WhatAmISupposedToPutHere — CHANGES_REQUESTED** [review]

File: asahi-btsync/src/main.rs (hunk 1 above)
Comment (jannau) [reply asahi-btsync/src/main.rs]: yes, copied and forgot about it, changed

File: asahi-btsync/src/main.rs (hunk 2 above)
Comment (jannau) [reply asahi-btsync/src/main.rs]: fixed, thanks

**Pull request merged**